addons/calendar/static/src/js/base_calendar.js

```diff
@@ -10,6 +15,154 @@ var WebClient = require('web.WebClient');
 
 var FieldMany2ManyTags = relational_fields.FieldMany2ManyTags;
 
+/**
+ * Calendar EventDeleteDialog
+ *
+ * This is the dialog called when the user try to delete a recurring event.
+ * It displays three buttons in order to delete one/future/all events. Once the
+ * user has choose the correct action the dialog return it to the calendar controller.
+ */
+
+var EventDeleteDialog = Dialog.extend({
+    template: "EventDeleteDialog",
+    events: _.extend({}, Dialog.prototype.events, {
+        'click .o_delete-event': '_onDeleteRecurrentEvents',
+    }),
+    /**
+     * @override
+     * @param {Widget} parent
+     * @param {interger} id calendar event id
+     */
+    init: function (parent, id){
+        //Button cancel that close the modal.
+        var button_cancel = {
+            text: _t("Cancel"),
+            classes: 'btn btn-default',
+            close: true,
+        };
+        this._super(parent, {
+            title: _t('This event is linked to a recurrence.'),
+            size: "medium",
+            buttons: [button_cancel],
+        });
+        this.id = id;
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Detect the action choosen by the user and return it to the CalendarController.
```

Comment:
I do not understand too much the usefulness of this commentary.

Trigger up the selected action (delete one/future/all events).
@ see _onDeleteRecurrentEvent